### PR TITLE
dont upload singles if coinc already uploaded

### PR DESCRIPTION
**Pull request by GarethCabournDavies**
A check for if a single event had been uploaded as a coincident event already (and therefore not upload the singles) was requested - I think this should work, but am not sure on where the tests are kept at the moment

### CODE REVIEW

File: bin/pycbc_live
Comment (titodalcanton) [inline bin/pycbc_live:425]: We probably want some tolerance on the merger time, maybe 100 ms? I would also `logging.info()` the fact that we are skipping an upload.

**Comment by GarethCabournDavies** [comment]
Do we have a recommended pycbc_live short test anywhere? I haven't tested what I have put in so though I think it should work I'm not really sure

**Comment by titodalcanton** [comment]
@GarethCabournDavies we have the usual example that runs in the CI tests, https://github.com/gwastro/pycbc/blob/master/examples/live/run.sh, although that does not have single-detector candidates enabled. You are currently the best person to create the necessary fit file though, so if you want to expand the test, that would be great.

**Comment by GarethCabournDavies** [comment]
I'll rebase this against master once #4000 is in, and then we can confirm that this is doing its job (one of the injections does currently produce both a single and a coinc)

**Comment by GarethCabournDavies** [comment]
Reminder that this is (I think) ready to go, just waiting on the #4000 tests to be merged and rebased against before setting this to merge

**Comment by GarethCabournDavies** [comment]
(N.B. Github has a horrible way to present commits when I've rebased to master)

**Comment by spxiwh** [comment]
@GarethCabournDavies I disabled auto-merge here. I think you need to resolve the rebase issue, as there are now changes in 99 files, suggesting this didn't rebase correctly.

**Comment by GarethCabournDavies** [comment]
> @GarethCabournDavies I disabled auto-merge here. I think you need to resolve the rebase issue, as there are now changes in 99 files, suggesting this didn't rebase correctly.

Ah, thanks - No idea where that messed up

**Review by titodalcanton — APPROVED** [review]

**Comment by GarethCabournDavies** [comment]
I just realised that the CI test isn't actually helping us check here as the file still gets saved, just not uploaded

I'll run another test just to double check this is working to prevent the upload, and if not then I'll try another PR

**Comment by titodalcanton** [comment]
@GarethCabournDavies was this actually tested with uploads enabled? I am hitting this error (with current master):
```
Traceback (most recent call last):
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/site-packages/mpi4py/__main__.py", line 7, in <module>
    main()
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/site-packages/mpi4py/run.py", line 198, in main
    run_command_line(args)
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/site-packages/mpi4py/run.py", line 47, in run_command_line
    run_path(sys.argv[0], run_name='__main__')
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/runpy.py", line 268, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/pycbc.live/.conda/envs/conda-test/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/pycbc.live/.conda/envs/conda-test/bin/pycbc_live", line 969, in <module>
    evnt.check_coincs(list(results.keys()), best_coinc,
  File "/home/pycbc.live/.conda/envs/conda-test/bin/pycbc_live", line 413, in check_coincs
    self.last_few_coincs_uploaded.append(event.merger_time)
AttributeError: 'numpy.ndarray' object has no attribute 'append'
```
Indeed, `last_few_coincs_uploaded` is initialized as a Numpy array which does not have an append method. Did you mean to initialize it as a list maybe?

**Comment by titodalcanton** [comment]
The `logging.info()` line also contains a bug 🤦 I need to stop reviewing code while moving!

**Comment by GarethCabournDavies** [comment]
Sorry! - I think I didn't enable uploads in the testing for a very good reason, but I can't remember what it was

**Pull request merged**
